src/api/spec/jobs/update_package_meta_job_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the actions
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe UpdatePackageMetaJob, type: :job, vcr: true do
+  include ActiveJob::TestHelper
+
+  let(:user) { create(:confirmed_user, login: 'tom') }
+  let(:project) { user.home_project }
+  let!(:package1) { create(:package, name: 'package_1', project: project) }
+  let!(:package2) { create(:package, name: 'package_2', project: project) }
+  let!(:package3) { create(:package, name: 'package_3', project: project) }
+  let!(:branch_package) { BranchPackage.new(project: project.name, package: package1.name) }
+
+  it 'is in default queue' do
+    expect(UpdatePackageMetaJob.new.queue_name).to eq('default')
+  end
+
+  describe '#perform' do
+    let!(:other_branch_package) { BranchPackage.new(project: project.name, package: package2.name) }
+
+    before do
+      User.current = user
+      branch_package.branch
+      bp = Package.last.update_backendinfo
+      bp.update_attributes(links_to_id: nil)
+    end
+
+    subject(:job) { UpdatePackageMetaJob.new.perform }
+
+    it 'should remove it' do
+      expect { subject }.to change { BackendPackage.count }.by(-1)
+    end
+  end
+
+  describe '#scan_links' do
+    # If the package has a link it will check if exist a BackendPackage, otherwise, it will create it.
+    before do
+      User.current = user
+      branch_package.branch
+    end
+
+    subject(:job) { UpdatePackageMetaJob.new.scan_links }
+
+    context 'with an updated BackendPackage' do
+      before do
+        # It's needed to create a BackendPackage entry
+        Package.last.update_backendinfo
+      end
+
+      it 'should exist a BackendPackage' do
+        job
+        expect(BackendPackage.count).to eq(1)
+      end
+    end
```

Comment:
I am not sure if this is a valid way of using a subject. At least it's quite confusing / hard to see where a method call is actually made.
Why don't you use a before blog?